subprojects/base-services/src/test/groovy/org/gradle/util/CollectionUtilsTest.groovy

```diff
@@ -126,8 +126,18 @@ class CollectionUtilsTest extends Specification {
         collect([1, 2, 3] as Object[], transformer { it * 2 }) == [2, 4, 6]
     }
 
-    def "collect iterable"() {
+    def "collect non-collection Iterable"() {
         expect:
+        def nonCollectionIterable = new Iterable() {
+            @Override
+            Iterator iterator() {
+                return [1, 2, 3].iterator()
+            }
+        }
```

Comment:
I think you can write this as `{ [1, 2, 3] } as Iterable`. Just mentioning because I think it's cool. :)

Reply:
but that's already an `Iterable`, that's the other branch, it's covered in the test below this one

Reply:
Changed it to `[iterator: { [1, 2, 3].iterator() }] as Iterable`